File: phaseII/palindrome.c
#include <stdbool.h>
bool isPalindrome(int x);

int main(void)
{
    
}

bool isPalindrome(int x){
    int y, copy;
    unsigned int z = 0;

    if (x < 0)
        return (false);
    copy = x;

    while (1)
    {
        y = x % 10;
        x /= 10;

        z = z * 10;
        z += y;

        if (x == 0)
                if (z == copy)
                    return (true);
                else
                    return (false);
    }
}

File: phaseII/palindrome.py
#!/usr/bin/python3

class Solution:
    def isPalindrome(x: int) -> bool:
        copy = x
        z = 0

        while True:
            y = x % 10
            x /= 10

            z *= 10
            z += y

            print(z)

            if x  == 0:
                if z == copy:
                    print("True")
                    return True
                else:
                    print("False")
                    return False

Solution.isPalindrome(4554)